src/modules/Elsa.Workflows.Core/Extensions/ActivityExecutionContextExtensions.cs

```diff
@@ -436,6 +438,22 @@ public static IEnumerable<ActivityExecutionContext> GetDescendants(this Activity
         }
     }
 
+    /// <summary>
+    /// Sets extension data in the metadata. Represents specific data that is exposed generically for an activity.
+    /// </summary>
+    public static void SetExtensionsMetadata(this ActivityExecutionContext context, Dictionary<string, object?> data)
+    {
+        context.Metadata[ExtensionsMetadataKey] = data;
+    }
```

Comment:
It would be better if this ExtensionsMetadata owns the dictionary, rather than relying on the caller to provide the dictionary, considering that other activities might also want to add extension data. If they were  to call this method, they will be overriding existing values.

Instead, let this extension method try to read an existing value at the ExtensionsMetadataKey, and if it doesn't exist, create it and add it to Metadata.

The SetExtensionsMetadata therefore should only have to receive a key and a value to store in the extensions dictionary.